master/internal/rm/kubernetesrm/kubernetes_resource_manager_intg_test.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+)
+
+const (
+	defaultNamespace   = "default"
+	auxNodeName        = "aux"
+	compNodeName       = "comp"
+	pod1NumSlots       = 4
+	pod2NumSlots       = 8
+	nodeNumSlots       = 8
+	nodeNumSlotsCPU    = 20
+	podServSlotTypeGPU = "randomDefault"
+)
+
+// Number of active slots on given nodes.
+var nodeToSlots = map[string]int{
+	auxNodeName:  pod1NumSlots,
+	compNodeName: pod2NumSlots,
+}
+
+func TestGetAgentsAndGetSlots(t *testing.T) {
+	// Create agent nodes.
+	auxNode := createMockNode(auxNodeName)
+	compNode := createMockNode(compNodeName)
+
+	nodes := make(map[string]*k8sV1.Node)
+	nodes[auxNode.Name] = &auxNode
+	nodes[compNode.Name] = &compNode
+
+	// Create resource managers with different device types for their respective pod service.
+	rmGPU := createMockRM(nodes, podServSlotTypeGPU)
+	rmCUDA := createMockRM(nodes, device.CUDA)
+	rmCPU := createMockRM(nodes, device.CPU)
+
+	rmList := []*ResourceManager{rmGPU, rmCUDA, rmCPU}
+	for _, rm := range rmList {
+		agentsResp, err := rm.GetAgents(&apiv1.GetAgentsRequest{})
+		require.NoError(t, err)
+		require.Equal(t, len(nodes), len(agentsResp.Agents))
+		for _, agent := range agentsResp.Agents {
+			_, ok := nodes[agent.Id]
+			require.True(t, ok)
+			agentResp, err := rm.GetAgent(&apiv1.GetAgentRequest{AgentId: agent.Id})
+			require.NoError(t, err)
+			require.Equal(t, agentResp.Agent.Id, agent.Id)
+
+			slotsResp, err := rm.GetSlots(&apiv1.GetSlotsRequest{AgentId: agent.Id})
+			require.NoError(t, err)
+			require.Equal(t, nodeNumSlots, len(slotsResp.Slots))
+			activeSlots := 0
+			cnt := 0
+			for _, slot := range slotsResp.Slots {
+				slotResp, err := rm.GetSlot(&apiv1.GetSlotRequest{
+					AgentId: agent.Id,
+					SlotId:  strconv.Itoa(cnt),
+				})
+				require.NoError(t, err)
+				slotID, err := strconv.Atoi(slotResp.Slot.Id)
+				require.NoError(t, err)
+				require.Equal(t, cnt, slotID)
+				if slot.Container != nil {
+					activeSlots++
+					require.True(t, slotResp.Slot.Container != nil)
+				}
+			}
+			require.Equal(t, nodeToSlots[agent.Id], activeSlots)
+		}
+	}
+
+	// At the moment, ROCm resource manager should panic when this device type is used in requests
+	// about cluster resources because ROCm is currently not supported in Kubernetes, so we check
+	// this resource manager separately.
+	require.Panics(t, testROCMResourceManager)
+}
+
+func testROCMResourceManager() {
+	auxNode := createMockNode(auxNodeName)
+	nodes := make(map[string]*k8sV1.Node)
+	nodes[auxNode.Name] = &auxNode
+	rm := createMockRM(nodes, device.ROCM)
+	rm.GetAgents(&apiv1.GetAgentsRequest{})
+}
+
+func createMockNode(name string) k8sV1.Node {
+	resourceList := createMockResourceList()
+	nodeStatus := k8sV1.NodeStatus{Allocatable: resourceList}
+	return k8sV1.Node{
+		ObjectMeta: metaV1.ObjectMeta{
+			ResourceVersion: "1",
+			Name:            name,
+		},
+		Status: nodeStatus,
+	}
```

Comment:
```suggestion
	return k8sV1.Node{
		ObjectMeta: metaV1.ObjectMeta{
			ResourceVersion: "1",
			Name:            name,
		},
		Status: k8sV1.NodeStatus{Allocatable: createMockResourceList()},
	}
```

Reply:
thought i commented this but i dont see it, so maybe i missed. for a lot of these helpers, i dont think the LoC saved is worth the indirection when someone is reading it. I would just inline the mock agent, resource list stuff.